lib/request.js

```diff
@@ -417,7 +417,13 @@ module.exports = {
 
   get subdomains() {
     var offset = this.app.subdomainOffset;
-    return (this.host || '')
+    var ip = /^(\d{1,3}\.){3}\d{1,3}$/gim;
```

Comment:
I don't think IPv6 is a big concern for the moment. However I did think about leaving a TODO in the comment, but, hey, using IPv4 address as the host is ugly already, who'd bother with IPv6 anyway.